internal/pool/pool.go

```diff
@@ -437,14 +438,23 @@ func (p *ConnPool) reaper(frequency time.Duration) {
 	ticker := time.NewTicker(frequency)
 	defer ticker.Stop()
 
-	for range ticker.C {
-		if p.closed() {
-			break
-		}
-		_, err := p.ReapStaleConns()
-		if err != nil {
-			internal.Logger.Printf("ReapStaleConns failed: %s", err)
-			continue
+	for {
+		select {
+		case <-ticker.C:
+			if p.closed() {
+				break
+			}
+			_, err := p.ReapStaleConns()
+			if err != nil {
+				internal.Logger.Printf("ReapStaleConns failed: %s", err)
+				continue
+			}
+		default:
+			if p.closed() {
+				break
+			}
+			// hand over execution rights to avoid tight loop
+			runtime.Gosched()
```

Comment:
This is still unnecesary tight (I guess). Could you add a `closedCh chan struct{}` that is `close(p.closed.Ch)` when pool is closed and listen on that channel here? Should be much more lightweight and we don't need `p.closed` checks anymore.

Reply:
resolved.